library/rsa.c

```diff
@@ -1679,11 +1619,8 @@ static int rsa_rsassa_pss_sign( mbedtls_rsa_context *ctx,
     int ret = MBEDTLS_ERR_ERROR_CORRUPTION_DETECTED;
     size_t msb;
 
-    RSA_VALIDATE_RET( ctx != NULL );
-    RSA_VALIDATE_RET( ( md_alg  == MBEDTLS_MD_NONE &&
-                        hashlen == 0 ) ||
-                      hash != NULL );
-    RSA_VALIDATE_RET( sig != NULL );
+    if( ( md_alg != MBEDTLS_MD_NONE || hashlen != 0 ) && hash == NULL )
```

Comment:
I would add additional test cases for each of these functions, to check both the `md_alg != MBEDTLS_MD_NONE` and `hashlen != 0` cases return `MBEDTLS_ERR_RSA_BAD_INPUT_DATA`.

Reply:
added new tests